### ISSUE TITLE
Triggering Debug View Hierarchy causes fatal error

### ISSUE THREAD

**Issue by stami** [groue/GRDB.swift#1159]
### What did you do?

1. Build and run demo app **GRDBAsyncDemo** or **GRDBCombineDemo** (does not reproduce with _GRDBDemoiOS_)
2. Click "Debug View Hierarchy"
3. View hierarchy capture fails due to error:
```
GRDB.swift-master/GRDB/Core/Statement.swift:27: Fatal error: Database was not used on the correct thread.
```

My guess is that it is related to `@Query` property wrapper as that is common in those crashing demo apps.

### What did you expect to happen?

View hierarchy capture succeed

### What happened instead?

Failure with the fatal error `Fatal error: Database was not used on the correct thread.`
<img width="336" alt="Screenshot 2022-01-29 at 11 30 07" src="https://user-images.githubusercontent.com/6566702/151655856-d04e992a-99a0-45eb-97d4-07335be3a8b6.png">

### Environment

**GRDB flavor(s):** GRDB, GRDBQuery
**GRDB version:** current `master`, also 5.17.0 that I run with my project
**Installation method:** Demo app straight from this repo, CocoaPods
**Xcode version:** 13.2.1
**Swift version:** 5.5
**Platform(s) running GRDB:** iOS tested
**macOS version running Xcode:** 12.1

### Demo Project

https://github.com/groue/GRDB.swift/tree/master/Documentation/DemoApps
- GRDBAsyncDemo
- GRDBCombineDemo

**NOT** _GRDBDemoiOS_

**Comment by groue** [comment]
Thanks for the report @stami 👍

**Comment by groue** [comment]
Fixed in v5.20.0
